File: src/com/leozhi/topic392/Solution05.java
package com.leozhi.topic392;

/**
 * @author leozhi
 * 双指针
 * 官方题解
 * 通过
 * 1ms
 */
class Solution05 {
    public boolean isSubsequence(String s, String t) {
        int n = s.length(), m = t.length();
        int i = 0, j = 0;
        while (i < n && j < m) {
            if (s.charAt(i) == t.charAt(j)) {
                i++;
            }
            j++;
        }
        return i == n;
    }
}

File: src/com/leozhi/topic88/Solution05.kt
package com.leozhi.topic88

/**
 * @author leozhi
 * 双指针
 * 通过
 * 212ms
 */
class Solution05 {
    fun merge(nums1: IntArray, m: Int, nums2: IntArray, n: Int) {
        var i = m
        var j = n
        var k = i-- + j-- + 1
        while (i >= 0 && j >= 0) {
            nums1[k--] = if(nums1[i] > nums2[j]) nums1[i--] else nums2[j--]
        }
        while (j >= 0) {
            nums1[k--] = nums2[j--]
        }
    }
}